src/App.js

```diff
@@ -1,16 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+  const [userColors, setUserColors] = useState({
+    local: '',
+    remote: '',
+  });
+
+  const toggleLiked = (id) => {
+    setChatData((chatData) =>
+      chatData.map((chat) => {
+        if (id === chat.id) {
+          return { ...chat, liked: !chat.liked };
+        } else {
+          return chat;
+        }
+      })
+    );
+  };
+
+  const calTotalLikes = (chatData) => {
+    return chatData.reduce((total, chat) => {
+      return chat.liked ? total + 1 : total;
+    }, 0);
+  };
+
+  const totalLikes = calTotalLikes(chatData);
+
+  const selectLocalColor = (color) => {
+    setUserColors((userColors) => {
+      return { ...userColors, local: color };
+    });
+  };
+
+  const selectRemoteColor = (color) => {
+    setUserColors((userColors) => {
+      return { ...userColors, remote: color };
+    });
+  };
```

Comment:
It would take refactoring of more than just these two methods, but how would you go about refactoring these into one method since they're nearly identical methods (except for the local and remote keys)?